include/soci/soci-platform.h

```diff
@@ -18,13 +25,63 @@
 #ifdef _MSC_VER
 #include <stdlib.h>
 
+//Disables warnings about STL objects need to have dll-interface and/or
+//base class must have dll interface
+#pragma warning(disable:4251 4275)
+
 // Define if you have the vsnprintf variants.
 #if _MSC_VER < 1500
 # define vsnprintf _vsnprintf
 #endif
 
+#if _MSC_VER >= 1400 //use secure version removing compiler warnings
+    #define sprintf sprintf_s //just use secure version( same prototype )
+    #define sscanf sscanf_s //just use secure version( same prototype )
+    #define strncpy strncpy_secure //use inline wrapper bellow
+    #define strcpy strcpy_secure //use inline wrapper bellow
+    #define snprintf snprintf_secure //use inline wrapper bellow
+
+    inline char* strncpy_secure(char* buf, const char* src, size_t count)
+    {
+        errno_t err = strncpy_s(buf,count,src,count);
+        if( err != 0 )
+            return 0;
+        return buf;
+    }
+
+    inline char* strcpy_secure(char* dest, const char* src)
+    {
+        errno_t err = strcpy_s(dest,strlen(src)+1,src);
```

Comment:
On 8 July 2015 at 10:57, ravselj <notifications@github.com> wrote:

> Looks like this can only be resolved correctly by defining
> _CRT_SECURE_NO_WARNINGS. Would that be better?
>

If you decide to use the preprocessor define(s), please add them to .h/cpp
files wherever necessary,
before any headers importing any 'insecure' function(s).
But, not to CMake or any build configuration.